Support arm64 with efi bootloader

woodrow-shen opened the pull request:
There is a arm64 system with efi boot and currently it failed to boot
uc18 due to ubuntu-core-rootfs of initramfs lost ${rootmnt}/boot/efi,
where it wasn't created from core18 snap. To consider this case, core18
should make sure /boot/efi can be built with the same hook.

sil2100 commented on hooks/016-ensure-bootloaders.chroot:
```diff
@@ -21,6 +22,5 @@ case "$(dpkg --print-architecture)" in
 EOF
         ;;
     amd64|i386)
-        mkdir /boot/efi
```
Can we simply remove the amd64|i386 case altogether now? Doesn't make sense to leave an empty case-statement here.

woodrow-shen replied:
@sil2100,
+1, I refreshed my branch again.

sil2100 reviewed (changes requested):
+1 on the overall changes, just requesting a small stylistic change inline.

I think this is a good approach, especially that we're unconditionally creating /boot/uboot already.

Commit:
Support arm64 with efi bootloader

There is a arm64 system with efi boot and currently it failed to boot
uc18 due to ubuntu-core-rootfs of initramfs lost ${rootmnt}/boot/efi,
where it wasn't created from core18 snap. To consider this case, core18
should make sure /boot/efi can be built with the same hook.

sil2100 reviewed (approved):
+1
